packages/hardhat-chai-matchers/src/properAddress.ts

```diff
@@ -0,0 +1,12 @@
+export function supportProperAddress(Assertion: Chai.AssertionStatic) {
+  Assertion.addProperty("properAddress", function (this: any) {
+    const subject = this._obj;
+    this.assert(
```

Comment:
I wonder if this shouldn't be split into two separate assertions, one asserting that it's the proper length, and the other asserting that it has the proper characters, so that we can tell the user "it's not the right length" or "it has non-hex characters" etc.

Reply:
Agree in principle, but tbh we are mainly adding these matchers for compatibility reasons. I don't think they are used a lot, so I wouldn't invest much more time on them.